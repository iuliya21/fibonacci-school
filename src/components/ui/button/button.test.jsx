import { render, fireEvent } from '@testing-library/react';
import { Button } from "./button";

test('Отрисовка компонента Button с текстом', () => {
  const { asFragment } = render(<Button text="Текст кнопки"/>);
  expect(asFragment()).toMatchSnapshot();
});

test('Отрисовка компонента Button без текста', () => {
  const { asFragment } = render(<Button />);
  expect(asFragment()).toMatchSnapshot();
});

test('Отрисовка заблокированного компонента Button', () => {
  const { asFragment } = render(<Button disabled />);
  expect(asFragment()).toMatchSnapshot();
});

test('Отрисовка компонента Button с индикацией загрузки', () => {
  const { asFragment } = render(<Button isLoader />);
  expect(asFragment()).toMatchSnapshot();
});

test('Клик по компоненту Button вызывает callback', () => {
  const onClickMock = jest.fn();
  const { getByRole } = render(<Button text="Текст кнопки" onClick={onClickMock} />);
  fireEvent.click(getByRole('button'));
  expect(onClickMock).toHaveBeenCalled();
});