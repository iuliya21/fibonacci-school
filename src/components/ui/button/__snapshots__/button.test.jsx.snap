// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button компонент Отрисовка заблокированного компонента Button 1`] = `
<DocumentFragment>
  <button
    class="text text_type_button text_color_primary button undefined false "
    disabled=""
    type="button"
  >
    <p
      class="text undefined"
    />
  </button>
</DocumentFragment>
`;

exports[`Button компонент Отрисовка компонента Button без текста 1`] = `
<DocumentFragment>
  <button
    class="text text_type_button text_color_primary button undefined false "
    type="button"
  >
    <p
      class="text undefined"
    />
  </button>
</DocumentFragment>
`;

exports[`Button компонент Отрисовка компонента Button с индикацией загрузки 1`] = `
<DocumentFragment>
  <button
    class="text text_type_button text_color_primary button undefined loader "
    disabled=""
    type="button"
  >
    <img
      alt="Загрузка."
      class="loader_icon"
      src="loader.svg"
    />
  </button>
</DocumentFragment>
`;

exports[`Button компонент Отрисовка компонента Button с текстом 1`] = `
<DocumentFragment>
  <button
    class="text text_type_button text_color_primary button undefined false "
    type="button"
  >
    <p
      class="text undefined"
    >
      Текст кнопки
    </p>
  </button>
</DocumentFragment>
`;
